tests/framework/extensions/cloudcredentials/linode/create.go

```diff
@@ -0,0 +1,29 @@
+package linode
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/cloudcredentials"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+)
+
+const linodeCloudCredNameBase = "linodeCloudCredential"
+
+// CreateLinodeCloudCredentials is a helper function that takes the rancher Client as a parameter and creates
+// a Linode cloud credential, and returns the CloudCredential response
+func CreateLinodeCloudCredentials(rancherClient *rancher.Client) (*cloudcredentials.CloudCredential, error) {
```

Comment:
if this is a method of the rancher client it should be with the rancher client.  I would accept the rancher client as a parameter instead of making this a method.

Reply:
This the design we have gone with, with the other node providers, I was just following the same design. Since we are maintaining our own `CloudCredential` struct we decided to maintain the different cloud credentials in our extensions